src/pty.c

```diff
@@ -0,0 +1,118 @@
+#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
+#include <errno.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <signal.h>
+#include <pty.h>
```

Comment:
on osx we need to include <util.h> instead

Reply:
Instead of `pty.h`? So:

```c
#ifdef DARWIN
#include <util.h>
#else
#include <pty.h>
#endif
```